### PR TITLE
[Tagger] Add a containerd collector and support container_env_as_tags

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

- Build a `containerd` tagger collector
- Support `container_env_as_tags`

### Motivation

Feature request

### Additional Notes

Will make the `docker` collector support `container_env_as_tags` in a separate PR.

### Describe how to test your changes

* The collector should be auto-enabled in a k8s containerd environment
* Configure some env as tags, example: `'{"env":"env", "foo":"bar", "POD_NAME":"custom_pod_name"}'`
```
agent tagger-list
...
=== Entity container_id://ff6da781abecee727221862630faded41769af6fbd2b5f1137fa1e218bd4bdea ===
== Source containerd ==
Tags: [bar:foo_val custom_pod_name:redis env:dev]
== Source kube-metadata-collector ==
Tags: []
== Source kubelet ==
Tags: [container_id:ff6da781abecee727221862630faded41769af6fbd2b5f1137fa1e218bd4bdea display_container_name:redis_redis env:dev image_name:redis/redis image_tag:latest kube_container_name:redis kube_namespace:default pod_name:redis pod_phase:running service:redis-svc short_image:redis version:v1]
...
===
```
* Kill and create multiple pods, make sure the `containerd` tagger collector detects the container lifecycle events and apply the `container_env_as_tags` config if it matches.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

### CODE REVIEW

File: pkg/tagger/collectors/docker_main.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (c *DockerCollector) Detect(_ context.Context, out chan<- []*TagInfo) (Coll
 
 	// We lower-case the values collected by viper as well as the ones from inspecting the labels of containers.
 	c.labelsAsTags = retrieveMappingFromConfig("docker_labels_as_tags")
-	c.envAsTags = retrieveMappingFromConfig("docker_env_as_tags")
+	c.envAsTags = retrieveMappingFromConfig("docker_env_as_tags") // TODO: Merge with container_env_as_tags
```
Comment (ahmed-mez) [inline pkg/tagger/collectors/docker_main.go:56]: Will do it in a separate PR

File: pkg/tagger/collectors/containerd_main.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+// processEvent handles container creation/update/deletion events and sends a TagInfo accordingly.
+func (c *ContainderdCollector) processEvent(ctx context.Context, event *containerdevents.Envelope) {
+	var handleCreateUpdate = func(id string) {
+		container, isPauseContainer, err := c.containerByID(ctx, id)
+		if err != nil {
+			log.Debugf("Could not fetch container %q: %w", id, err)
+			return
+		}
+
+		if isPauseContainer {
```
Comment (juliogreff) [inline pkg/tagger/collectors/containerd_main.go:148]: I wonder if we shouldn't emit a `TagInfo` with no tags for such cases, as a `tagger.Tag` call about a pause container will always result in a fetch (which is relatively expensive) that never returns anything, although we know that a pause container will never have any tags.

Comment (ahmed-mez) [reply pkg/tagger/collectors/containerd_main.go:148]: Why would we call `tagger.Tag` with an entity corresponding to to a pause container? Are you thinking of a particular tagger client that could do it?

File: pkg/tagger/collectors/docker_main.go (hunk 1 above)
Comment (ahmed-mez) [reply pkg/tagger/collectors/docker_main.go:56]: https://github.com/DataDog/datadog-agent/pull/9056

File: pkg/tagger/collectors/containerd_main.go (hunk 2 above)
Comment (juliogreff) [reply pkg/tagger/collectors/containerd_main.go:148]: No, I don't have a particular one in mind. Ideally they shouldn't, I agree. Just monitoring for spikes in entities not found and fixing it at call sites is a reasonable way to go, too, so all in all this LGTM as it is 🚀

**Review by juliogreff — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -1940,6 +1940,13 @@ api_key:
 #   <NAMESPACE_LABEL>: <TAG_KEY>
 #   <HIGH_CARDINALITY_NAMESPACE_LABEL_NAME>: +<TAG_KEY>
 
+## @param container_env_as_tags - map - optional
```
Comment (ruthnaebeck) [inline pkg/config/config_template.yaml:1943]: If there's an equivalent environment variable, we are now adding those below the `@param`. For example:

```
## @env DD_CONTAINER_ENV_AS_TAGS - format - optional
```

**Review by ruthnaebeck — COMMENTED** [review]
Docs review
